File: Apis/Domain/Entities/UserTeam.cs
using System;
using System.Collections.Generic;

namespace Domain.Entities;

public class UserTeam : BaseEntity
{

    public ulong? IsLeader { get; set; }

    public int? Teamid { get; set; }

    public int? Userid { get; set; }

    public virtual Team? Team { get; set; }

    public virtual User? User { get; set; }
}

File: EXE201_Heureux/src/main/java/com/example/exe201_heureux/entity/UserTeam.java
package com.example.exe201_heureux.entity;

import jakarta.persistence.*;
import lombok.*;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Entity
@Table(name = "User_Team")
public class UserTeam {
    @Id
    @Column(name = "id", nullable = false)
    private Integer id;

    @ManyToOne(fetch = FetchType.LAZY, cascade = CascadeType.DETACH)
    @JoinColumn(name = "userid")
    private User userid;

    @ManyToOne(fetch = FetchType.LAZY, cascade = CascadeType.DETACH)
    @JoinColumn(name = "teamid")
    private Team teamid;

    @Column(name = "isLeader")
    private Boolean isLeader;

}